os/net/mac/mac-sequence.c

```diff
@@ -45,79 +45,89 @@
 #include <string.h>
 
 #include "contiki-net.h"
+#include "lib/random.h"
 #include "net/mac/mac-sequence.h"
 #include "net/packetbuf.h"
 
-struct seqno {
-  linkaddr_t sender;
-  clock_time_t timestamp;
-  uint8_t seqno;
-};
+/* Log configuration */
+#include "sys/log.h"
+#define LOG_MODULE "mac-sequence"
+#define LOG_LEVEL LOG_LEVEL_MAC
 
 #ifdef NETSTACK_CONF_MAC_SEQNO_MAX_AGE
 #define SEQNO_MAX_AGE NETSTACK_CONF_MAC_SEQNO_MAX_AGE
 #else /* NETSTACK_CONF_MAC_SEQNO_MAX_AGE */
-#define SEQNO_MAX_AGE (20 * CLOCK_SECOND)
+#define SEQNO_MAX_AGE (20 /* seconds */)
 #endif /* NETSTACK_CONF_MAC_SEQNO_MAX_AGE */
 
-#ifdef NETSTACK_CONF_MAC_SEQNO_HISTORY
-#define MAX_SEQNOS NETSTACK_CONF_MAC_SEQNO_HISTORY
-#else /* NETSTACK_CONF_MAC_SEQNO_HISTORY */
-#define MAX_SEQNOS 16
-#endif /* NETSTACK_CONF_MAC_SEQNO_HISTORY */
-static struct seqno received_seqnos[MAX_SEQNOS];
+#define BEFORE(a, b) ((int16_t)((a) - (b)) < 0)
+
+struct deduplication_record {
+  uint8_t last_broadcast_seqno;
```

Comment:
I think this layout pretty much guarantees padding, starting with the largest members and finishing with the smallest should pad less.

I don't know if the decreased readability is worth it though, some maintainer should chime in on this.

Reply:
Things will become more simple if we agree on dropping duplicate detection for broadcast frames.

Reply:
I adjusted the order in the struct and added some #ifdef's for eliminating the broadcast deduplication. Am not sure if we need that and if it should be on or off by default.